### PR TITLE
Allow main vs. ref comparision to use a cached main in addition to a cached ref analysis run

### LINKED ISSUE #345: Allow comparison run from cached main and ref runs
Allow a comparison (main vs. ref) analysis run to be made from cached main and ref analysis runs, not just cached ref runs.  This would allow:
* running the main analysis on a different machine from the one where the comparison is run (e.g. if it's more convenient to run the comparison on the machine housing the ref. run)
* save a lot of time on the comparison run if the main vs. obs run is desired (which it nearly always is)

### PR DESCRIPTION
**Pull request by xylar**
Adds a config option `mainRunConfigFile`.  By pointing to a config file for analysis that has already run
(from which only the output directory and subdirectories are used), the climatologies, time series, mapping files and masks can be reused rather than needlessly recomputing them.

This merge also fixes a string vs. byte issue that occurs when indexing depths in Argo and SOSE climatology maps.  An error occurs when trying to reuse depth data (possibly only in python 3) because byte strings don't compare correctly with unicode strings.

closes #345

### CODE REVIEW

**Comment by xylar** [comment]
@darincomeau, I thought you might be a good person to test this out since you've been running "main vs ref" type jobs quite a lot lately.  Let me know if it's unclear how this would work.  I'll add some documentation right away, so watch for another commit in the next few minutes.

**Comment by darincomeau** [comment]
@xylar this looks like a great feature to have - thanks for doing this. As part of the review do you want me to test this on a pair of runs?

**Comment by xylar** [comment]
Yes, that would be great!

**Comment by darincomeau** [comment]
@xylar I'm running tests now from this branch and got the following error during one of the potential density tasks:

Plotting 2-d maps of Sea Floor Potential Density climatologies for ANN on the antarctic grid...
analysis task climatologyMapSchmidtko_plotPotentialDensity_ANN_antarctic failed during run
Traceback (most recent call last):
  File "mpas_analysis/shared/analysis_task.py", line 320, in run
    self.run_task()
  File "mpas_analysis/ocean/plot_climatology_map_subtask.py", line 398, in run_task
    remappedRefClimatology.depthSlice.values):
  File "/global/project/projectdirs/acme/software/anaconda_envs/edison/base/envs/e3sm_unified_1.2.4_py2.7_nox/lib/python2.7/site-packages/xarray/core/common.py", line 176, in __getattr__
    (type(self).__name__, name))
AttributeError: 'Dataset' object has no attribute 'depthSlice'


Do you know what might be causing this?

**Comment by xylar** [comment]
Hmm, maybe easiest if I come by and we look into it together.  I'd have to look at the data set it's trying to read to see why there's no `depthSlice` variable in that file.

File: config.example [hunk 1]
```diff
@@ -47,6 +47,13 @@ preprocessedReferenceRunName = None
 # reference run is desired.
 # referenceRunConfigFile = /path/to/config/file
 
+# config file for a main run on which the analysis was already run to
```
[the remark this answers is not in the document]
Comment (xylar) [reply config.example]: `preprocessedReferenceRunName` refers to a preprocessed E3SM v0 run that @milenaveneziani has made available on some machines.  I agree that this name could be changed but it is unrelated to the "control" vs. "main" runs.

**Comment by xylar** [comment]
Although I have not finished testing to be sure, I strongly suspect that the error above:
```
AttributeError: 'Dataset' object has no attribute 'depthSlice'
```
is related to the fact that @darincomeau's config file is pointing to an outdated observations directory on Edison.  @milenaveneziani, should we delete that directory or move it somewhere the average user wont' find it to prevent this from happening in the future?  There are similar outdated directories on other systems.  We might need to work with @vanroekel to delete some of them, since I think he owns them (e.g. on Theta and Anvil).

**Comment by milenaveneziani** [comment]
No, I would not delete the old observations/ directory anywhere.. This is because it contains much more data than the newer directory, like the original data that we used for computing climatologies, for example.
I would just move it or rename it for now, until we have a better plan on how to proceed (we have talked about this before but we never really decide on a way forward).

**Comment by xylar** [comment]
Okay, sounds reasonable.  @milenaveneziani, would you be willing to rename:
`acme/observations` --> `acme/old_observations`
`acme/mpas_analysis` --> `acme/old_mpas_analysis_region_and_mapping_files`
Or something like that?

**Comment by xylar** [comment]
Actually, I own `mpas_analysis` so I guess I should be the one to rename that one.

**Comment by milenaveneziani** [comment]
sure. I would add a bit more description to the old_obs folder, maybe something like `old_observations_with_original_data`. I know it's long, but at least nobody will be confused about its content.

**Comment by xylar** [comment]
Yep, sounds great!

**Comment by vanroekel** [comment]
Could you summarize what you'd like me to do on theta?  I see I own that directory, but don't know why.  Could I chown them to either you (@xylar) or @milenaveneziani ?

**Comment by xylar** [comment]
@darincomeau, I ran successfully with your config file!  No error on potential density plots that I can see.  I'll see if I can find out why...

**Comment by milenaveneziani** [comment]
@vanroekel: let me try this and I'll let you know if I run into permission problems.

**Comment by xylar** [comment]
@vanroekel, we would like you to rename them something like "old_observations" and/or "old_mpas_analysis_data" or just delete them if you think it's safe.  Everyone should be using the `diagnostics/observations` and `diagnostics/mpas_analysis` directories that are pointed to in example config files rather than the old directories from the Wild West days of MPAS-Analysis when everyone copied the missing analysis data from everywhere else.

**Comment by xylar** [comment]
I'm going to go ahead and merge this because I don't think @darincomeau's error is related to changes made here and I can't reproduce it right now.

**Comment by vanroekel** [comment]
okay, I have renamed my owned mpas_analysis_data directory on theta.

**Comment by darincomeau** [comment]
@xylar I agree my error is most likely a user error and not related to this PR. I updated the ocean obs path and re-ran the analysis, and still got this error.

**Comment by xylar** [comment]
@darincomeau, which python environment are you using?  I'm using the latest E3SM-unified (1.2.4).

**Comment by xylar** [comment]
Never mind, I can see from your error message above that you're using the same.
